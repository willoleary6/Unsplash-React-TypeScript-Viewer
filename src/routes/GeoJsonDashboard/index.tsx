import React from "react";
import { GeoMap } from "../../features/GeoMap";
export function GeoJsonDashboard(): JSX.Element {
    return (
        <>
            {/* Wrap it in a dummy jsx parent */}
            <div className="row">
                <div className="col-lg-4 mt-5"></div>
                <div className="col-lg-4 mt-5">
                    <GeoMap />
                </div>
                <div className="col-lg-4 mt-5"></div>
            </div>
        </>
    );
}
