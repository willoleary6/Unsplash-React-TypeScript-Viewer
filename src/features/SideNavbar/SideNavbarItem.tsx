import React from "react";
import { Link, useRouteMatch } from "react-router-dom";

import { Route } from "../../routes/routes";

interface SideNavbarItemProps {
    route: Route;
}

export const SideNavbarItem = ({ route }: SideNavbarItemProps): JSX.Element => {
    const match = useRouteMatch({
        path: route.path,
        exact: true,
    });

    return (
        <li className={match ? "active" : ""}>
            <Link to={route.path}>
                <>
                    {route.icon} <span className="nav-label">{route.label}</span>
                </>
            </Link>
        </li>
    );
};
