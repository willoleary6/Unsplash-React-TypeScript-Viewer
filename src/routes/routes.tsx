import React from "react";
import { GeoJsonDashboard } from "./GeoJsonDashboard";
import { UnsplashViewerDashboard } from "./UnsplashViewerDashboard";
/**
 * Route information
 *
 * Top level routes should always contain an icon and either a component or
 * subroutes. Second level routes should not contain an icon or subroutes but
 * should contain a component.
 */
export interface Route {
    id: string;
    path: string;
    label: string;
    icon?: JSX.Element;
    component?: React.FC;
    subroutes?: Route[];
}

export const routes: Route[] = [
    // Top level route with no subroutes
    {
        id: "GeoJsonRoute",
        path: "/",
        label: "GeoJSON",
        icon: <i className="fa fa-globe"></i>,
        component: GeoJsonDashboard,
    },
    {
        id: "UnsplashViewerDashboard",
        path: "/UnspashViewer",
        label: "Unsplash Viewer",
        icon: <i className="fa fa-picture-o"></i>,
        component: UnsplashViewerDashboard,
    },
];
