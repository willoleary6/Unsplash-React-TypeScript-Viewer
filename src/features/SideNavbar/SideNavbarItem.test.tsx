import React from "react";
import { SideNavbarItem } from "./SideNavbarItem";
import Enzyme, { mount } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { routes } from "../../routes/routes";
import { BrowserRouter as Router } from "react-router-dom";
Enzyme.configure({ adapter: new Adapter() });

describe("SideNavbarItem render testing", () => {
    it("renders panel and takes snapshot to track jsx element changes, will also test for exceptions in element code", () => {
        // using mount here instead of shallow because we want to render the child components too
        const wrapper = mount(
            <Router>
                <SideNavbarItem route={routes[0]} />
            </Router>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
