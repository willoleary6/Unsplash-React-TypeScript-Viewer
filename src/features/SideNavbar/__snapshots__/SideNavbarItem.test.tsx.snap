// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavbarItem render testing renders panel and takes snapshot to track jsx element changes, will also test for exceptions in element code 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SideNavbarItem
      route={
        Object {
          "component": [Function],
          "icon": <i
            className="fa fa-globe"
          />,
          "id": "GeoJsonRoute",
          "label": "GeoJSON",
          "path": "/",
        }
      }
    >
      <li
        className="active"
      >
        <Link
          to="/"
        >
          <LinkAnchor
            href="/"
            navigate={[Function]}
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-globe"
              />
               
              <span
                className="nav-label"
              >
                GeoJSON
              </span>
            </a>
          </LinkAnchor>
        </Link>
      </li>
    </SideNavbarItem>
  </Router>
</BrowserRouter>
`;
